### PR TITLE
Improve match summary stats

### PR DESCRIPTION
**Pull request by Gael-de-Sailly**
I'm trying to improve the formspec displaying the match summary. I've added a flag image with the color of the winner team, and some information (total kills, score, duration…) at the top of the window.
screenshot_20180417_091906

This may need more testing before being merged, in particular I'd like to see how it behaves on small screens. Also, feel free to give your opinion about window layout or code quality.

### CODE REVIEW

**Comment by rubenwardy** [comment]
Oh nice! Have you checked that this doesn't change /rankings?

**Comment by Gael-de-Sailly** [comment]
I've checked it but not sure I did with the latest version of my code.

Le 17/04/2018 à 00:58, rubenwardy a écrit :

File: mods/ctf_stats/gui.lua [hunk 1]
```diff
@@ -10,21 +10,64 @@ function ctf_stats.get_formspec_match_summary(stats)
...
+	local time_display = ""
+	if time >= 3600 then
+		time_display = math.floor(time/3600) .. "h"
+	end
+	time_display = time_display .. math.floor((time % 3600) / 60) .. "m" .. math.floor(time % 60) .. "s"
+	ret = ret .. "label[8,0;Duration]"
+	ret = ret .. "label[10,0;" .. time_display .. "]"
+	ret = ret .. "label[8,0.5;Total score]"
+	ret = ret .. "label[10,0.5;" .. math.floor(total_score*10)/10 .. "]"
+	-- END added by me
```
Comment (rubenwardy) [inline mods/ctf_stats/gui.lua]: No need for these comments

File: mods/ctf_stats/gui.lua [hunk 2]
```diff
@@ -10,21 +10,64 @@ function ctf_stats.get_formspec_match_summary(stats)
...
-function ctf_stats.get_formspec(title, players)
+function ctf_stats.get_formspec(title, players, header)
 	table.sort(players, function(one, two)
 		return one.score > two.score
 	end)
 
-	local ret = "size[12,6.5]"
-	ret = ret .. "vertlabel[0,0;" .. title .. "]"
+	local ret = "size[12,"..6.5+header.."]"
+	ret = ret .. "vertlabel[0,"..header..";" .. title .. "]"
```
Comment (rubenwardy) [inline mods/ctf_stats/gui.lua]: Please use a container instead.

```lua
local ret = "size[12,"..6.5+header.."]"
ret = ret .. "container[0," .. header .. "]"

-- stuff here

ret = ret .. "container_end[]"
return ret
```

you then don't need to modify any other tags

**Comment by rubenwardy** [comment]
This partially fixes #151, it's just missing per-team info like score and kills

**Comment by rubenwardy** [comment]
You have an error:

```
Checking mods/ctf_stats/gui.lua                   1 warning
    mods/ctf_stats/gui.lua:14:8: variable 'game_won' is never accessed
```

File: mods/ctf_stats/gui.lua (hunk 1 above)
Comment (Gael-de-Sailly) [reply mods/ctf_stats/gui.lua]: Oh yes, quick comments I've forgotten to remove (indeed, just after submitting the PR, I realized that I've submitted it too fast)

File: mods/ctf_stats/gui.lua (hunk 2 above)
Comment (Gael-de-Sailly) [reply mods/ctf_stats/gui.lua]: Will try this, thank you

**Comment by ClobberXD** [comment]
Great improvements! :+1: 

Now that this PR is open, we might as well add more detailed stats (like separate column for bounty) before merging...

**Pull request merged**